### PR TITLE
Add privacy sensitive mode for API 30+

### LINKED ISSUE #1093: Support overriding input stream privacy sensitivity
Android 11 added [AAudioStreamBuilder_setPrivacySensitive()](https://developer.android.com/ndk/reference/group/audio#aaudiostreambuilder_setprivacysensitive) for AAudio. This API allows to override default privacy sensitivity of input stream which is otherwise based on input preset.

Oboe's [AudioStreamBuilder](https://github.com/google/oboe/blob/master/include/oboe/AudioStreamBuilder.h) should have the corresponding property.

When specified, it shall override default privacy sensitivity. When not specified, it shall not invoke the above AAudio API and thus lets the system to decide the privacy sensitivity.

### PR DESCRIPTION
**Pull request by robertwu1**
setPrivacySensitive/isPrivacySensitive are APIs in AAudio for overriding the default input presets. We are adding them to Oboe here.

Instead of using a boolean like in AAudio, Oboe is using a tristate of Unspecified/Enabled/Disabled. This allows a default implementation to be used and a fallback pattern for OpenSLES. Under the hood, AAudio also uses a tristate for this.

Fixes #1093

### CODE REVIEW

File: src/aaudio/AAudioLoader.h [hunk 1]
```diff
@@ -141,7 +150,7 @@ class AAudioLoader {
 
     typedef int32_t (*signature_I_PSKPLPL)(AAudioStream *, clockid_t, int64_t *, int64_t *);
 
-    typedef bool    (*signature_B_PS)(AAudioStream *);
+    typedef bool    (*signature_O_PS)(AAudioStream *);
```
Comment (philburk) [inline src/aaudio/AAudioLoader.h:153]: Thanks for spotting this.

File: tests/testStreamOpen.cpp [hunk 2]
```diff
@@ -460,3 +460,69 @@ TEST_F(StreamOpenOutput, OboeExtensions){
         ASSERT_TRUE(OboeExtensions::isMMapEnabled());
     }
 }
+
+TEST_F(StreamOpenInput, AAudioInputSetPrivacySensitiveModeUnspecifiedMedia){
```
Comment (philburk) [inline tests/testStreamOpen.cpp:464]: Should name say Unprocessed?

File: tests/testStreamOpen.cpp [hunk 3]
```diff
@@ -460,3 +460,69 @@ TEST_F(StreamOpenOutput, OboeExtensions){
...
+    if (getSdkVersion() >= __ANDROID_API_R__){
```
Comment (philburk) [inline tests/testStreamOpen.cpp:465]: You could run these tests on <R and then conditionally choose different expected results. Then we can check if it causes crashes.

**Review by philburk — CHANGES_REQUESTED** [review]

File: tests/testStreamOpen.cpp (hunk 3 above)
Comment (robertwu1) [reply tests/testStreamOpen.cpp:465]: Added a separate test for <R below. Adding more if statements makes the test confusing

File: tests/testStreamOpen.cpp [hunk 4]
```diff
@@ -526,3 +526,13 @@ TEST_F(StreamOpenInput, OpenSLESInputSetPrivacySensitiveModeGetsUnspecified){
     ASSERT_EQ(mStream->getPrivacySensitiveMode(), PrivacySensitiveMode::Unspecified);
     ASSERT_TRUE(closeStream());
 }
+
+TEST_F(StreamOpenInput, OldAndroidVersionInputSetPrivacySensitiveModeGetsUnspecified){
+    if (getSdkVersion() < __ANDROID_API_R__) {
+        mBuilder.setDirection(Direction::Input);
+        mBuilder.setInputPreset(InputPreset::VoiceCommunication);
```
Comment (philburk) [inline tests/testStreamOpen.cpp:533]: If you try to set privacy when <R does it crash?

**Review by philburk — CHANGES_REQUESTED** [review]

File: tests/testStreamOpen.cpp (hunk 4 above)
Comment (robertwu1) [reply tests/testStreamOpen.cpp:533]: No, it does not crash. It simply ignores the parameter.

**Review by philburk — CHANGES_REQUESTED** [review]

File: tests/testStreamOpen.cpp (hunk 4 above)
Comment (robertwu1) [reply tests/testStreamOpen.cpp]: Oops. Nice catch!

**Pull request merged**